app/services/verify_link_service.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class VerifyLinkService < BaseService
+  def call(field)
+    @link_back = ActivityPub::TagManager.instance.url_for(field.account)
+    @url       = field.value
+
+    perform_request!
+
+    return unless link_back_present?
+
+    field.mark_verified!
+    field.account.save!
+  end
+
+  private
+
+  def perform_request!
+    @body = Request.new(:get, url).add_headers('Accept' => 'text/html').perform do |res|
+      res.code != 200 ? nil : res.body_with_limit
+    end
+  end
+
+  def link_back_present?
+    return false if @body.empty?
+
+    Nokogiri::HTML(@body).xpath('//link[@rel="me"]').any? { |link| link['href'] == @link_back }
```

Comment:
Shouldn't the XPath read `//link[@rel="me"]|//a[@rel="me"]`? The copy you've written mentions `<a rel="me"` but this won't match that.

Reply:
My bad, it should be `a` instead of `link`, not sure if `link` is worth checking at all, that would be a real rarity

Reply:
True, both ``link`` and ``a`` are valid for ``rel=me``.

Reply:
I initially thought it was a “link” attribute and not a “a” attribute. Can't find the spec (if any) right now, but if link's in it, you should restore it, even if it “would be a real rarity”.